### ISSUE TITLE
Add simple spinner to page to show while loading JS

### ISSUE THREAD

**Issue by noisecapella** [mitodl/micromasters#1739]

**Comment by singingwolfboy** [comment]
Can we clarify this issue a bit? Does [`static/js/components/Loader.js`](https://github.com/mitodl/micromasters/blob/master/static/js/components/Loader.js) resolve it, or is there more work to be done? If we want a separate spinner to display before all the JS has finished loading, then can we make it look similar to that spinner?

**Comment by pdpinch** [comment]
I think this was already done?

![screenshot 2016-12-13 08 08 21](https://cloud.githubusercontent.com/assets/430126/21141472/616b1884-c10b-11e6-8b43-e607e7b9bf73.png)

**Comment by singingwolfboy** [comment]
That's why I'm seeking clarification, here. There are several phases to the loading process:

1. Client requests HTML of page.
2. Client receives HTML and identifies other referenced assets, such as images, CSS, and Javascript. Client requests those assets, in parallel as much as possible.
3. Client receives Javascript and starts executing it. Javascript instructs client to contact API endpoints. Client makes XHR requests, in parallel as much as possible.
4. Client has fully loaded all necessary data, and displays the dashboard (or whatever should be displayed).

I don't think it's possible to show a spinner between phase 1 and 2. We _could_ show a spinner between phase 2 and 3, if the spinner was implemented in HTML rather than in Javascript. We already have a spinner between phase 3 and 4, as shown above.

Is this issue advocating that we should show a spinner between phase 2 and 3? If so, should it be visually identical to the spinner shown between phase 3 and 4, or should it look different?

**Comment by noisecapella** [comment]
I think we can show a spinner between 1 and 2. We can load a small bundle of javascript which only shows a spinner, then show that spinner while loading and running the rest of the javascript. We could even just do an animated gif at that stage

I'm not really sure this is worth it though, I'm having trouble getting this to show up on my computer. Maybe I'm just on a connection that's too fast?
